src/AutoRest.CSharp/LowLevel/Extensions/ExampleValueSnippets.cs

```diff
@@ -376,12 +380,12 @@ private static ValueExpression GetExpressionForObjectType(ObjectType objectType,
                 if (valueDict.TryGetValue(property.InputModelProperty!.SerializedName, out var exampleValue))
                 {
                     properties.Remove(property);
-                    argument = GetExpression(propertyType, exampleValue, property.SerializationFormat, includeCollectionInitialization: true);
+                    argument = GetExpression(propertyType, exampleValue, SerializationBuilder.GetSerializationFormat(exampleValue.Type), includeCollectionInitialization: true);
```

Comment:
Similar problem - `ObjectTypeProperty` may or may not represent a property from specification, it may have different type or even belong to a different object in case of grouping or flattening. That's why I would prefer `GetExpressionForObjectType` to use `SerializationObjectType.JsonSerialization` as a source for object initializer.